README.md
# PISDA
PISDA - Provide Instant SQLite Data Access.

This program allow to work with SQLite3 database, from console, or CSharp,<br>
also, this can open SQLite3 databases, and execute SQL-requests for this.


+ System.Data.SQLite.dll			- library [System.Data.SQLite.dll](https://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki)
+ PISDA.cs										- PISDA, which allow to working with any databases, using System.Data.SQLite.dll, placed near this.
+ Compile_PISDA.bat						- ".bat"-file to compile PISDA.cs, with System.Data.SQLite.dll, placed near this file.
+ PISDA.exe										- output executable file, created after run "Compile_PISDA.bat". If this not exists, this will be created.



Changes.txt
________________________________________________________________________________________________________________
Fresh changes - in the top of this text file.
________________________________________________________________________________________________________________
Download https://system.data.sqlite.org/downloads/1.0.115.5/sqlite-netFx46-static-binary-bundle-x64-2015-1.0.115.5.zip
Unpack.
Rename "System.Data.SQLite.dll" to "sqlite3.dll"

BUILD:
	Windows 10 (.NET 4.8):
		csc /out:PISDA_w10.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_w10.exe
							OK!
					Linux, mono:
						mono PISDA_w10.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_w10.exe
							System.BadImageFormatException
					
	Linux, mono:
		csc /out:PISDA_mono.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_mono.exe
							OK!
					Linux, mono:
						mono PISDA_mono.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_mono.exe
							System.BadImageFormatException

	Windows XP (.NET 4.0):
		csc /out:PISDA_XP.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_XP.exe
							OK!
					Linux, mono:
						mono PISDA_XP.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_XP.exe
							System.BadImageFormatException
________________________________________________________________________________________________________________
https://github.com/username1565/PISDA/tree/PISDA_linux_mono
it's the same as https://github.com/username1565/PISDA/tree/407374c42acb8a6015edd55336ca29b9b4056099
but file PISDA.cs was been rewrited with using "Mono.Data.Sqlite", instead of "System.Data.SQLite",
and this can working with "Mono.Data.Sqlite.dll", instead of "System.Data.SQLite.dll".

"Mono.Data.Sqlite.dll" was been added there, too.

This can be compiled on linux, using mono, as:
	csc /out:PISDA.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
or as
	./build_mono.sh

Also, this can be runned on linux, using mono, as:
	mono PISDA.exe
or as
	./mono_run_with_sql.sh
	
After compilation "PISDA.exe", on linux (using mono),
this can be runned on Windows (with .NET Framework 4.0), too,
but probram return throw exception, about "sqlite3.dll" not found.

"sqlite3.dll" - it's something like "SQLite.Interop.dll",
and this contains inside "System.Data.SQLite.dll",
and "Mono.Data.Sqlite.dll" using this from "sqlite3.dll".
So "System.Data.SQLite.dll" was been renamed to "sqlite3.dll",
and after this, "PISDA.exe", working on Windows (with .NET Framework 4.0).

On Windows XP (.NET 4.0), "PISDA_linux_mono" can be sucessfully compilled and runned (with "Mono.Data.Sqlite.dll", and "sqlite3.dll"), too.

But after try to run "PISDA.exe", compiled on linux (using mono),
and after try to run this on Windows 10 (with .NET Framework 4.8, instead of .NET Framework 4.0),
	C:\Windows\Microsoft.NET\Framework\v4.0.30319\csc.exe --version
		-> 4.8.4084.0
This program is not working on windows.

________________________________________________________________________________________________________________
Add "build_mono.sh"
Add "mono_run_with_sql.sh"
________________________________________________________________________________________________________________
Add "Mono.Data.Sqlite.dll" from this source: https://dll-overhaul.com/file/mono.data.sqlite%20(2).dll/45133.html
Add "Mono.Data.Sqlite.dll-source.txt" with hashes, and description of library.

BUILD:
	Windows 10 (.NET 4.8):
		csc /out:PISDA_w10.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_w10.exe
							System.BadImageFormatException
					Linux, mono:
						mono PISDA_w10.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_w10.exe
							OK!
					
	Linux, mono:
		csc /out:PISDA_mono.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_mono.exe
							System.BadImageFormatException
					Linux, mono:
						mono PISDA_mono.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_mono.exe
							OK!

	Windows XP (.NET 4.0):
		csc /out:PISDA_XP.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_XP.exe
							System.BadImageFormatException
					Linux, mono:
						mono PISDA_XP.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_XP.exe
							OK!
							
Seems like something not found in Windows 10 (.NET 4.8).
________________________________________________________________________________________________________________
Seems, like mono's "Mono.Data.Sqlite.dll" is not complete...
________________________________________________________________________________________________________________
Rename "System.Data.SQLite.dll" to "sqlite3.dll"

BUILD:
	Windows 10 (.NET 4.8):
		csc /out:PISDA_w10.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_w10.exe
							System.TypeInitializationException: Инициализатор типа "Mono.Data.Sqlite.UnsafeNativeMethods" выдал исключение. ---> System.BadImageFormatException
					Linux, mono:
						mono PISDA_w10.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_w10.exe
							MissingMethodException. Method not found: "System.String System.String.Format(System.IFormatProvider, System.String, System.Object)".					
					
	Linux, mono:
		csc /out:PISDA_mono.exe /reference:Mono.Data.Sqlite.dll PISDA.cs
			OK!
				RUN:
					Windows 10 (.NET 4.8):
						PISDA_mono.exe
							System.TypeInitializationException: Инициализатор типа "Mono.Data.Sqlite.UnsafeNativeMethods" выдал исключение. ---> System.BadImageFormatException
					Linux, mono:
						mono PISDA_mono.exe
							OK!
					Windows XP (.NET 4.0):
						PISDA_mono.exe
							MissingMethodException. Method not found: "System.String System.String.Format(System.IFormatProvider, System.String, System.Object)".					

	Windows XP (.NET 4.0):

build_mono.sh
#	To build PIDSA.exe on linux, using mono, install "mono", and run:
#./build_mono.sh

# On linux, instead of System.Data.SQLite.dll, program is using Mono.Data.SQLite.dll.
# This DLL contains, near PISDA.exe.
# PISDA.exe can be runned with this file, on linux, using mono.

#	Compile PIDSA.exe witn "Mono.Data.SQLite.dll"
csc /out:PISDA.exe /reference:Mono.Data.Sqlite.dll PISDA.cs

#	run PISDA.exe with Mono, on linux:
mono PISDA.exe

# After try to run on windows "PISDA.exe", which compiled on linux, with "Mono.Data.SQLite.dll",
#	program return throw-exception, about "sqlite3.dll" not found.
#	"sqlite3.dll" it's renamed "System.Data.SQLite.dll", because there is contains "SQLite.Interop.dll", inside.
#	This file need to run "PIDSA.exe" with "Mono.Data.SQLite.dll", on windows,
#	but filename of this file must to be "sqlite3.dll", to provide access there, from "Mono.Data.SQLite.dll", on windows.
#		Copy "System.Data.SQLite.dll" to "sqlite3.dll".
cp "System.Data.Sqlite.dll" "sqlite3.dll"

#	Now, PISDA.exe can be runned on windows, with .NET Framework 4.0. On Windows10, there is throw-exception, because .NET Framework 4.8 there.